core/src/main/java/io/questdb/griffin/SqlCodeGenerator.java

```diff
@@ -242,6 +267,133 @@ private RecordCursorFactory createFullFatAsOfJoin(
         return master;
     }
 
+    @NotNull
+    private RecordCursorFactory createFullFatLtJoin(
```

Comment:
This is similar to asof join. Can we make factory constructor a functional interface? These factories take the same args. This way we can remove code duplication